### ISSUE TITLE
Move settings from classes/catalogue/*.php to configuration

### ISSUE THREAD

**Issue by nichtich** [pkiraly/qa-catalogue-web#108]
Thinks like name and catalogue link should be configurable in `configuration.cnf` so there is no need to create PHP code for each new library.

**Comment by nichtich** [comment]
With [this commit](https://github.com/pkiraly/qa-catalogue-web/commit/6786e5d2b9d52f98409abd723d8ccfb72dfe8066) the only method implemented for specific catalogues is `getOpacLink` (see `grep -o 'function .*' classes/catalogue/*.php | grep -v Catalogue.php`). We could replace the PHP classes with configuration files, but I'd first like to switch / add JSON file format for configuration.

**Comment by pkiraly** [comment]
I agree with the basic idea, but I am afraid it can not be fully implemented. Now the blocker is `getOpacLink`. In the majority of the catalogues the templating solution is working, but in other cases there is a complex function, e.g.:

ÖNB:
```php
  function getOpacLink($id, $record) {
    foreach ($record->getFields('035') as $tag35) {
      if (isset($tag35->subfields->a)) {
        $tag35a = $tag35->subfields->a;
        if (preg_match('/^\(AT-OBV\)/', $tag35a)) {
          $id = preg_replace('/^\(AT-OBV\)/', '', $tag35a);
          break;
        } else if (!preg_match('/^\(/', $tag35a)){
          $id = $tag35a;
        }
      } else {
        error_log('no 035$a in ' . json_encode($tag35));
      }
    }
    return 'http://data.onb.ac.at/rec/' . trim($id);
  }
```
NKP:
```PHP
  function getOpacLink($id, $record) {
    $rid = FALSE;
    foreach ($record->getFields('015') as $tag) {
      if (isset($tag->subfields->a)) {
        $rid = str_replace('cnb', '', $tag->subfields->a);
        break;
      }
    }
    if ($rid === FALSE)
      $rid = preg_replace('/^[a-z]{1,3}\d{4}/', '00', trim($id));
    return 'https://aleph.nkp.cz/F/?func=direct'
         . '&doc_number=' . $rid
         . '&local_base=CNB';
  }
```
So, the solution should also work with classes. The other important advantage of classes is that we can store information there about the catalogue, so the config based solution should also support this, I mean we should be able to store the configuration files in the repository.

**Comment by nichtich** [comment]
I agree we cannot fully get rid of PHP classes for catalogues, but the number could be reduced. 

> The other important advantage of classes is that we can store information there about the catalogue

That's the point I want to avoid. Information should better be stored in JSON files instead of PHP classes so the information can easier be read and written automatically. The same information should not be stored in multiple repositories and files but there should be one source of truth for thinks like name and URL of a catalogue.

We can postpone this for a later version to support JSON config files instead or in addition to `configuration.cnf`. Maybe this can be aligned with the JSON format used at <https://github.com/pkiraly/qa-catalogue-uitest/tree/main/catalogues>.
